crates/core/src/network/networking.rs

```diff
@@ -67,7 +67,8 @@ impl NetworkSender for Networking {
     {
         let target_is_generational = to.is_generational();
         let mut attempts = 0;
-        let mut retry_policy = self.connect_retry_policy.iter();
+        let retry_policy = self.options.connect_retry_policy.clone();
+        let mut retry_policy = retry_policy.iter();
```

Comment:
Why do you need to clone here? I think if you capture max_attempts() here you wouldn't have a problem with the projection reference below.

Reply:
Ah definitely, my fault. for some reason in my head iter() consumed `self` but it actually doesn't.